client/components/mobile-phone-change/mobile-phone-change.js

```diff
@@ -35,17 +31,46 @@ class MobilePhoneChange extends React.Component {
     };
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleChange = this.handleChange.bind(this);
-    this.validateToken = this.validateToken.bind(this);
   }
 
   async componentDidMount() {
-    await this.validateToken();
+    const {
+      cookies,
+      orgSlug,
+      logout,
+      verifyMobileNumber,
+      setUserData,
+    } = this.props;
+    let {userData} = this.props;
+    const isValid = await validateToken(
+      cookies,
+      orgSlug,
+      setUserData,
+      userData,
+    );
+    if (isValid) {
+      ({userData} = this.props);
+      const {phone_number} = userData;
+      this.setState({phone_number});
+      verifyMobileNumber(true);
+    } else {
+      logout(cookies, orgSlug);
```

Comment:
I am not convinced we should add this kind of logic in every component, looks duplicated

Reply:
Actually, on the status page, we have to handle session logout for the captive portal also. That's why I though to handle logout from the component itself.

Reply:
we can provide a callable which can default to this action, then if any component needs to customize it can pass a more complex callable